trade-finance-manager-api/api-tests/api.ts

```diff
@@ -0,0 +1,101 @@
+import { AnyObject } from '@ukef/dtfs2-common';
+import request from 'supertest';
+import dotenv from 'dotenv';
+import { IncomingHttpHeaders } from 'http';
+import { TestUser } from './types/test-user';
+import { TestApi } from './types/test-api';
+import { File } from './types/file';
+
+dotenv.config();
+
+const { TFM_API_KEY } = process.env;
+
+const getHeaders = (token?: string): IncomingHttpHeaders => {
+  const headers: IncomingHttpHeaders = {
+    'content-type': 'application/json',
+    'x-api-key': TFM_API_KEY,
+  };
+  if (token) {
+    headers.Authorization = token;
+  }
+
+  return headers;
+};
+
+export const createApi = (app: unknown): TestApi => ({
+  as: (user: TestUser) => {
+    const token = user?.token ? user.token : '';
+
+    return {
+      post: (data: AnyObject) => ({
+        to: async (url: string) => request(app).post(url).send(data).set(getHeaders(token)),
+      }),
+
+      postEach: (list: AnyObject[]) => ({
+        to: async (url: string) => {
+          const results = [];
+
+          for (const data of list) {
+            const result = await request(app).post(url).send(data).set(getHeaders(token));
+
+            results.push(result);
+          }
+
+          return results;
+        },
+      }),
+
+      put: (data: AnyObject) => ({
+        to: async (url: string) => request(app).put(url).send(data).set(getHeaders(token)),
+      }),
+
+      putMultipartForm: (data: AnyObject, files: File[] = []) => ({
+        to: async (url: string) => {
+          const apiRequest = request(app).put(url).set(getHeaders(token));
+
+          if (files.length) {
+            await Promise.all(files.map((file) => apiRequest.attach(file.fieldname, file.filepath)));
+          }
+
+          await Promise.all(
+            Object.entries(data).map(([fieldname, value]) => {
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              return apiRequest.field(fieldname, value as any);
```

Comment:
Are we able to remove the eslint disable and `any`?

Reply:
Thanks @MarRobSoftwire good finds - I would vote for no.2 and definitely not no.1 🙈 

I know it bloats the code, but it's better than disabling eslint rules, what do you think?

Conscious that we already have a lot of eslint disabling (e.g search for `@typescript-eslint/no-unsafe-argument` and click through each file), really keen to _not_ add anymore such disables